lib/puppet/ssl/host.rb

```diff
@@ -394,6 +394,37 @@ def state
 
   private
 
+  # Load a previously generated CSR either from memory or from disk
+  # @return [Puppet::SSL::CertificateRequest, nil]
+  def load_certificate_request
+    if Puppet::SSL::CertificateRequest.indirection.terminus_class == :memory
+      return Puppet::SSL::CertificateRequest.indirection.find(cert_name)
+    end
+
+    request_path = certificate_request_location(name)
+    if Puppet::FileSystem.exist?(request_path)
+      Puppet::SSL::CertificateRequest.from_s(Puppet::FileSystem.read(request_path))
+    end
+  end
+
+  # Submit the CSR to the CA, either via an HTTP PUT request, or when testing,
+  # via the indirector. This also caches a copy of the CSR on disk.
```

Comment:
It looks like there another unmentioned case when it uses the indirector

Reply:
when do we stop using the indirector when testing?

Reply:
When we remove the Ruby CA, see https://tickets.puppetlabs.com/browse/PUP-8912 (I don't know exactly when we will do that, some time in Puppet 6 horizon). All of the tests that use the `memory` and `ca` terminii are CA-related.